packages/apps/tools/src/hooks/use-modules-query.ts

```diff
@@ -0,0 +1,57 @@
+import type {
+  ChainwebChainId,
+  ChainwebNetworkId,
+} from '@kadena/chainweb-node-client';
+import { CHAINS } from '@kadena/chainweb-node-client';
+import { listModules } from '@kadena/client-utils';
+import { useQuery } from '@tanstack/react-query';
+
+const transformModules = (
+  data: string[][],
+  networkId: ChainwebNetworkId,
+  chainIds?: ChainwebChainId[],
+) => {
+  const chains = chainIds ?? CHAINS;
+
+  return data.flatMap((x, index) => {
+    return x.map((y) => {
+      return {
+        name: y,
+        chainId: chains[index],
+        networkId,
+      };
+    });
+  });
+};
+
+const fetchModules = async (
+  networkId: ChainwebNetworkId,
+  chainIds?: ChainwebChainId[],
+) => {
+  const chains = chainIds ?? CHAINS;
+
+  const promises = chains.map((chainId) =>
+    listModules({
+      defaults: { networkId, meta: { chainId } },
+    }),
+  );
+
+  const results = await Promise.all(promises);
```

Comment:
Do we need some kind of error handling?

Reply:
It can be taken care of at the "consumer" side of things, since this function is used in Tanstack's `useQuery`, this would then  e.g. result in an error on `pages/modules/explorer/index.tsx`;

```
const {error, data}  = useModulesQuery('mainnet01');

if (error) {
   console.error(error.message)
}
```

Reply:
Awesome! 

 I missed the useModuleQuery below..  makes a lot of sense now.